iac-common/src/test/java/org/sonar/iac/common/api/tree/impl/TextRangesTest.java

```diff
@@ -56,6 +58,19 @@ void test_merge_no_range() {
       .withMessage("Can't merge 0 ranges");
   }
 
+  @Test
+  void test_mergeRange() {
+    TestHasTextRange testRange1 = new TestHasTextRange(1,2, 3, 4);
+    TestHasTextRange testRange2 = new TestHasTextRange(5,6, 7, 8);
```

Comment:
Maybe you can also checkout Jonas test case which avoids having a dedicated Test class.